static/src/javascripts/projects/common/modules/ui/smartAppBanner.js

```diff
@@ -58,7 +60,16 @@ define([
             fullTemplate = tmp + (detect.getBreakpoint() === 'mobile' ? '' : tablet);
 
         msg.show(template(fullTemplate, DATA[platform.toUpperCase()]));
+
         cookies.add(COOKIE_IMPRESSION_KEY, impressions + 1);
+
+        fastdom.read(function () {
+            var $banner = $('.site-message--ios, .site-message--android');
+            var bannerHeight = $banner.dim().height;
+            if (window.scrollY !== 0) {
+                window.scrollTo(window.scrollX, window.scrollY + bannerHeight);
+            }
+        });
```

Comment:
I wasn't sure if deferring the scroll update to the next animation frame would result in a flicker, but in my testing it seems fine.